src/mono/wasm/runtime/loader/globals.ts

```diff
@@ -1,36 +1,51 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+/* eslint-disable @typescript-eslint/triple-slash-reference */
+/// <reference path="../types/side-car.d.ts" />
+
 import { exceptions, simd } from "wasm-feature-detect";
 
 import gitHash from "consts:gitHash";
 
-import type { AssetEntryInternal, GlobalObjects, LoaderHelpers, RuntimeHelpers } from "../types/internal";
+import type { AssetEntryInternal, DotnetModuleInternal, GlobalObjects, LoaderHelpers, MonoConfigInternal, RuntimeHelpers } from "../types/internal";
 import type { MonoConfig, RuntimeAPI } from "../types";
 import { assert_runtime_running, is_exited, is_runtime_running, mono_exit } from "./exit";
 import { assertIsControllablePromise, createPromiseController, getPromiseController } from "./promise-controller";
 import { mono_download_assets, resolve_single_asset_path, retrieve_asset_download } from "./assets";
 import { setup_proxy_console } from "./logging";
 import { invokeLibraryInitializers } from "./libraryInitializers";
-import { hasDebuggingEnabled } from "./config";
+import { deep_merge_config, hasDebuggingEnabled } from "./config";
 import { logDownloadStatsToConsole, purgeUnusedCacheEntriesAsync } from "./assetsCache";
 
+// if we are the first script loaded in the web worker, we are expected to become the sidecar
```

Comment:
so this indicates that emscripten pthreads will have their first loaded script be emscripten's threadmain equivalent, which installs onmessage. and when we spawn a sidecar we will load our script in it first instead to boot the runtime there, which will set this property?

Reply:
yes, emscripten is loading `dotnet.native.worker.js` and that installs `onMessage`
yes, we will be loading `dotnet.js` as first script of the sidecar

Reply:
maybe state this more explicitly in the comment, but it does at least feel somewhat evident. the expectation that emscripten installs onmessage could become broken somehow, I guess, maybe if they start using addEventListener instead?